public/app/plugins/datasource/cloudwatch/datasource.js

```diff
@@ -191,7 +185,20 @@ function (angular, _) {
           });
         }
 
-        return this.getDimensionValues(region, namespace, metricName, dimensions);
+        return this.getDimensionValues(region, namespace, metricName, dimensions).then(function(result) {
+          return _.map(result, function(dimensions) {
+            var values = _.chain(dimensions)
+            .sortBy(function(dimension) {
+              return dimension.Name;
+            })
+            .map(function(dimension) {
+              return dimension.Name + '=' + dimension.Value;
```

Comment:
To help understanding, I add screen capture of current implementation.
![templating_editor](https://cloud.githubusercontent.com/assets/224552/11142879/e337e988-8a33-11e5-8275-3f20edac8e28.png)
![dimension_values](https://cloud.githubusercontent.com/assets/224552/11142883/ea128f42-8a33-11e5-9a5a-25aa3ae22763.png)

For ElastiCache Redis, we get like following values.
CacheClusterId=cluster-001
CacheClusterId=cluster-001,CacheNodeId=0001
CacheClusterId=cluster-002
CacheClusterId=cluster-002,CacheNodeId=0001